HID Sensors framework support
Support for HID Sensor framework. This creates a MFD instance
for a sensor hub and identifies all the sensors connected to it.
Each sensor is registered as a MFD cell, so that sensor specific
processing can be done in a separate driver. Each sensor
drivers can use the service provided by this driver to register
for events and handle data streams. Each sensor driver can format
data and present to user mode using input or IIO interface.
Depends: HID && HAS_IOMEM
---
--- /var/tmp/portage/sys-kernel/gentoo-sources-5.19.17/image/usr/src/linux-5.19.17-gentoo//.config.orig	2022-10-31 13:57:59.705695527 -0400
+++ /var/tmp/portage/sys-kernel/gentoo-sources-5.19.17/image/usr/src/linux-5.19.17-gentoo//.config	2022-11-01 16:20:03.187610331 -0400
@@ -7127,8 +7127,7 @@
 CONFIG_HID_ZEROPLUS=m
 CONFIG_ZEROPLUS_FF=y
 CONFIG_HID_ZYDACRON=m
-CONFIG_HID_SENSOR_HUB=m
-# CONFIG_HID_SENSOR_CUSTOM_SENSOR is not set
+# CONFIG_HID_SENSOR_HUB is not set
 CONFIG_HID_ALPS=m
 CONFIG_HID_MCP2221=m
 # end of Special HID drivers
@@ -7761,7 +7760,6 @@
 #
 # HID Sensor RTC drivers
 #
-# CONFIG_RTC_DRV_HID_SENSOR_TIME is not set
 # CONFIG_RTC_DRV_GOLDFISH is not set
 CONFIG_DMADEVICES=y
 # CONFIG_DMADEVICES_DEBUG is not set
@@ -8406,7 +8404,6 @@
 CONFIG_DMARD10=m
 # CONFIG_FXLS8962AF_I2C is not set
 # CONFIG_FXLS8962AF_SPI is not set
-CONFIG_HID_SENSOR_ACCEL_3D=m
 CONFIG_IIO_CROS_EC_ACCEL_LEGACY=m
 CONFIG_IIO_ST_ACCEL_3AXIS=m
 CONFIG_IIO_ST_ACCEL_I2C_3AXIS=m
@@ -8548,8 +8545,6 @@
 #
 # Hid Sensor IIO Common
 #
-CONFIG_HID_SENSOR_IIO_COMMON=m
-CONFIG_HID_SENSOR_IIO_TRIGGER=m
 # end of Hid Sensor IIO Common
 
 #
@@ -8659,7 +8654,6 @@
 CONFIG_FXAS21002C=m
 CONFIG_FXAS21002C_I2C=m
 CONFIG_FXAS21002C_SPI=m
-CONFIG_HID_SENSOR_GYRO_3D=m
 CONFIG_MPU3050=m
 CONFIG_MPU3050_I2C=m
 CONFIG_IIO_ST_GYRO_3AXIS=m
@@ -8689,7 +8683,6 @@
 CONFIG_DHT11=m
 # CONFIG_HDC100X is not set
 CONFIG_HDC2010=m
-CONFIG_HID_SENSOR_HUMIDITY=m
 CONFIG_HTS221=m
 CONFIG_HTS221_I2C=m
 CONFIG_HTS221_SPI=m
@@ -8750,8 +8743,6 @@
 # CONFIG_SENSORS_ISL29018 is not set
 # CONFIG_SENSORS_ISL29028 is not set
 # CONFIG_ISL29125 is not set
-CONFIG_HID_SENSOR_ALS=m
-# CONFIG_HID_SENSOR_PROX is not set
 # CONFIG_JSA1212 is not set
 CONFIG_RPR0521=m
 CONFIG_LTR501=m
@@ -8793,7 +8784,6 @@
 CONFIG_BMC150_MAGN_I2C=m
 # CONFIG_BMC150_MAGN_SPI is not set
 # CONFIG_MAG3110 is not set
-CONFIG_HID_SENSOR_MAGNETOMETER_3D=m
 # CONFIG_MMC35240 is not set
 CONFIG_IIO_ST_MAGN_3AXIS=m
 CONFIG_IIO_ST_MAGN_I2C_3AXIS=m
@@ -8815,8 +8805,6 @@
 #
 # Inclinometer sensors
 #
-CONFIG_HID_SENSOR_INCLINOMETER_3D=m
-CONFIG_HID_SENSOR_DEVICE_ROTATION=m
 # end of Inclinometer sensors
 
 #
@@ -8831,7 +8819,6 @@
 #
 # Linear and angular position sensors
 #
-CONFIG_HID_SENSOR_CUSTOM_INTEL_HINGE=m
 # end of Linear and angular position sensors
 
 #
@@ -8866,7 +8853,6 @@
 CONFIG_IIO_CROS_EC_BARO=m
 # CONFIG_DLHL60D is not set
 # CONFIG_DPS310 is not set
-# CONFIG_HID_SENSOR_PRESS is not set
 # CONFIG_HP03 is not set
 CONFIG_ICP10100=m
 CONFIG_MPL115=m
@@ -8919,7 +8905,6 @@
 #
 CONFIG_LTC2983=m
 CONFIG_MAXIM_THERMOCOUPLE=m
-CONFIG_HID_SENSOR_TEMP=m
 CONFIG_MLX90614=m
 CONFIG_MLX90632=m
 # CONFIG_TMP006 is not set
